dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/FieldResource.java

```diff
@@ -35,36 +29,78 @@
 import com.dotmarketing.util.UtilMethods;
 import com.liferay.portal.model.User;
 
+import static com.dotcms.util.CollectionsUtils.imap;
+
 @Path("/v1/contenttype/{typeId}/fields")
 public class FieldResource implements Serializable {
 	private final WebResource webResource;
+	private final FieldAPI fieldAPI;
 
 	public FieldResource() {
-		this(new WebResource());
+		this(new WebResource(), APILocator.getContentTypeFieldAPI());
 	}
 
 	@VisibleForTesting
-	public FieldResource(final WebResource webresource) {
+	public FieldResource(final WebResource webresource, final FieldAPI fieldAPI) {
+		this.fieldAPI = fieldAPI;
 		this.webResource = webresource;
 	}
 
 	private static final long serialVersionUID = 1L;
 
 
+	@PUT
+	@JSONP
+	@NoCache
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+	public Response updateFields(@PathParam("typeId") final String typeId, final String fieldsJson,
+										   @Context final HttpServletRequest req) throws DotDataException, DotSecurityException {
+
+		final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+		final User user = initData.getUser();
+		
+		Response response = null;
+		
+		try {
+			final List<Field> fields = new JsonFieldTransformer(fieldsJson).asList();
+
+			for (Field field : fields) {
+				fieldAPI.save(field, user);
+			}
+
+			final List<Field> contentTypeFields = fieldAPI.byContentTypeId(typeId);
+			response = Response.ok(new ResponseEntityView(new JsonFieldTransformer(contentTypeFields).mapList())).build();
+		} catch (DotStateException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)